### PR TITLE
Change hard coded object names and uuids

### PR DESCRIPTION
**Pull request by Xenwar**
Change hard coded object names and uuids
fixes: https://github.com/metal3-io/cluster-api-provider-metal3/issues/592

### CODE REVIEW

**Review by lentzi90 — APPROVED** [review]
Fantastic! This makes it so much easier to follow what's going on in the tests!

**Comment by Xenwar** [comment]
> Fantastic! This makes it so much easier to follow what's going on in the tests!

Thanks for suggesting the idea https://github.com/metal3-io/cluster-api-provider-metal3/pull/563#discussion_r831825842

**Review by kashifest — APPROVED** [review]

**Comment by smoshiur1237** [comment]
Overall looks good, minor change needed in naming.

File: baremetal/suite_test.go [hunk 1]
```diff
@@ -48,14 +48,25 @@ var k8sClient client.Client
 var testEnv *envtest.Environment
 
 const (
-	clusterName       = "baremetal_testCluster"
-	metal3ClusterName = "baremetal_testmetal3Cluster"
-	namespaceName     = "baremetalns"
-	m3muid            = "11111111-9845-4321-1234-c74be387f57c"
-	bmhuid            = "22222222-9845-4c48-9e49-c74be387f57c"
+	clusterName            = "baremetal_testCluster"
```
Comment (Xenwar) [inline baremetal/suite_test.go:51]: replace _ by -
lowercase all names

Comment (Xenwar) [reply baremetal/suite_test.go:51]: done

File: controllers/suite_test.go [hunk 2]
```diff
@@ -53,12 +53,15 @@ var testEnv *envtest.Environment
...
-	metal3machineTemplateName = "controllerns_testmetal3machinetemplate"
+	clusterName            = "controller_testCluster"
+	metal3ClusterName      = "controller_testmetal3Cluster"
+	machineName            = "controller_testMachine"
+	metal3machineName      = "controller_testmetal3machine"
+	namespaceName          = "controller"
+	metal3DataTemplateName = "controller_testmetal3datatemplate"
+	baremetalhostName      = "controller_testbaremetalhostname"
+	metal3DataName         = "baremetal-metal3dataname"
+	metal3DataClaimName    = "baremetal-metal3dataclaim"
```
[the remark this answers is not in the document]
Comment (Xenwar) [reply controllers/suite_test.go:64]: done, thanks

File: controllers/suite_test.go [hunk 3]
```diff
@@ -53,12 +53,15 @@ var testEnv *envtest.Environment
...
-	namespaceName             = "controllerns"
-	metal3machineTemplateName = "controllerns_testmetal3machinetemplate"
+	clusterName            = "controller_testCluster"
+	metal3ClusterName      = "controller_testmetal3Cluster"
+	machineName            = "controller_testMachine"
+	metal3machineName      = "controller_testmetal3machine"
+	namespaceName          = "controller"
+	metal3DataTemplateName = "controller_testmetal3datatemplate"
+	baremetalhostName      = "controller_testbaremetalhostname"
+	metal3DataName         = "baremetal-metal3dataname"
```
[the remark this answers is not in the document]
Comment (Xenwar) [reply controllers/suite_test.go:63]: done, thanks.

File: controllers/suite_test.go [hunk 4]
```diff
@@ -53,12 +53,15 @@ var testEnv *envtest.Environment
...
-	metal3ClusterName         = "controllerns_testmetal3Cluster"
-	machineName               = "controllerns_testMachine"
-	metal3machineName         = "controllerns_testmetal3machine"
-	namespaceName             = "controllerns"
-	metal3machineTemplateName = "controllerns_testmetal3machinetemplate"
+	clusterName            = "controller-testcluster"
+	metal3ClusterName      = "controller-testmetal3cluster"
+	machineName            = "controller-testmachine"
+	metal3machineName      = "controller-testmetal3machine"
+	namespaceName          = "controller"
```
[the remark this answers is not in the document]
Comment (Xenwar) [reply controllers/suite_test.go:60]: Thanks, done

File: controllers/suite_test.go [hunk 5]
```diff
@@ -53,12 +53,15 @@ var testEnv *envtest.Environment
...
+	metal3DataTemplateName = "controller-testmetal3datatemplate"
+	baremetalhostName      = "controller-testbaremetalhostname"
+	metal3DataName         = "controller-metal3dataname"
```
[the remark this answers is not in the document]
Comment (Xenwar) [reply controllers/suite_test.go:63]: Thanks, done

**Review by smoshiur1237 — APPROVED** [review]
/lgtm

**Pull request merged**